sqlmesh/utils/pydantic.py

```diff
@@ -185,3 +192,73 @@ def wrapper(cls: t.Type, v: t.Any, values: t.Any, *args: t.Any, **kwargs: t.Any)
         return func(cls, v, values_dict, *args, **kwargs)
 
     return wrapper
+
+
+T = t.TypeVar("T")
+
+
+class PydanticTypeProxy(t.Generic[T]):
```

Comment:
[Nit] Shouldn't this definition be under `else` block since it's only relevant to pydantic v1?